Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs

```diff
@@ -93,8 +107,18 @@ endRideTransaction driverId bookingId ride mbFareParams mbRiderDetailsId merchan
             driver <- SQP.findById referredDriverId >>= fromMaybeM (PersonNotFound referredDriverId.getId)
             sendNotificationToDriver driver.merchantId FCM.SHOW Nothing FCM.REFERRAL_ACTIVATED referralTitle referralMessage driver.id driver.deviceToken
           Nothing -> pure ()
-  now <- getCurrentTime
-  let rideDate = getCurrentDate now
+
+  transporterConfig <- SCT.findByMerchantId merchantId >>= fromMaybeM (TransporterConfigNotFound merchantId.getId)
+  let govtCharges = fromMaybe 0 newFareParams.govtCharges
+  let (platformFee, cgst, sgst) = case newFareParams.fareParametersDetails of
+        DFare.ProgressiveDetails _ -> (0, 0, 0)
+        DFare.SlabDetails fpDetails -> (fromMaybe 0 fpDetails.platformFee, fromMaybe 0 fpDetails.cgst, fromMaybe 0 fpDetails.sgst)
+  let totalDriverFee = govtCharges + platformFee + round cgst + round sgst
```

Comment:
bro lets add as highPrec values and then at the end do round? this can introduce errors by removing decimal values from gst components.